api/v1alpha1/deployment_manager_order_types.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeploymentManagerOrderSpec defines the desired state of deployment manager template.
+type DeploymentManagerOrderSpec struct {
+	// Name is the name of the requested deployment manager.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Template is the name of the deployment manager template that will be used to create the
+	// deployment manager.
+	//
+	// +kubebuilder:validation:Required
+	Template string `json:"template,omitempty"`
```

Comment:
I don't think we need `omitempty` if the field is Required.

Reply:
OK, will remove it.